loader/loader.go

```diff
@@ -787,3 +788,29 @@ func convertVolumePath(volume types.ServiceVolumeConfig) types.ServiceVolumeConf
 	volume.Source = convertedSource
 	return volume
 }
+
+func resolveEnvironment(dict map[string]any, opts *Options) {
+	for key, value := range dict {
+		if key == "environment" {
+			if environment, ok := value.([]any); ok {
+				serviceEnvironment := []any{}
+				for _, env := range environment {
+					env, ok := env.(string)
+					if !ok {
+						continue
+					}
+					if found, ok := opts.Interpolate.LookupValue(env); ok {
```

Comment:
`Interpolate` being involved here is a bit confusing. I understand the actual goal is to access environment map, which `LookupValue` captured in options. So this basically is a hack - We should better add an explicit `Environment` (or `Variables` ?) attribute to `loader.Options` so we make it explicit we rely on the `ConfigDetails.Environment` here (and maybe in other places in future)